packages/components/src/components/carousel/carousel-utils.ts

```diff
@@ -113,14 +108,27 @@ export const isInfinitePagination = (amountOfPages: number): boolean => {
   return amountOfPages > INFINITE_BULLET_THRESHOLD;
 };
 
-export const renderPagination = (paginationEl: HTMLElement, amountOfPages: number, activeIndex: number): void => {
+export const renderPagination = (
+  paginationEl: HTMLElement,
+  amountOfPages: number,
+  activeIndex: number,
+  splide: Splide
+): void => {
   if (paginationEl) {
     // sanitize in case of removal of slide since activeIndex is from before splide.refresh()
     activeIndex = activeIndex > amountOfPages - 1 ? amountOfPages - 1 : activeIndex;
     paginationEl.innerHTML = Array.from(
       Array(amountOfPages),
       (_, i) => `<span class="bullet${i === activeIndex ? ' ' + bulletActiveClass : ''}"></span>`
     ).join('');
+
+    const bullets = paginationEl.querySelectorAll('.bullet');
+    bullets.forEach((bullet, i) => {
+      bullet.addEventListener('click', () => {
```

Comment:
Would be better to have single click listener on the parent and then figure out which bullet was clicked in the handler.